generic3g/ComponentSpecParser.F90

```diff
@@ -177,6 +188,61 @@ subroutine to_typekind(typekind, attributes, rc)
          _RETURN(_SUCCESS)
       end subroutine to_typekind
 
+      subroutine to_VerticalDimSpec(vertical_dim_spec, attributes, rc)
+         type(VerticalDimSpec) :: vertical_dim_spec
+         class(YAML_Node), intent(in) :: attributes
+         integer, optional, intent(out) :: rc
+
+         integer :: status
+         character(:), allocatable :: vertical_str
+
+         vertical_dim_spec = VERTICAL_DIM_NONE ! GEOS default
+         
+         if (.not. attributes%has('vertical_dim_spec')) then
+            _RETURN(_SUCCESS)
+         end if
+         call attributes%get(vertical_str, 'vertical_dim_spec', _RC)
+
+         select case (vertical_str)
+         case ('vertical_dim_none')
+            vertical_dim_spec = VERTICAL_DIM_NONE
+         case ('vertical_dim_center')
+            vertical_dim_spec = VERTICAL_DIM_CENTER
+         case ('vertical_dim_edge')
+            vertical_dim_spec = VERTICAL_DIM_EDGE
+         case default
+            _FAIL('Unsupported typekind')
+         end select
+
+         _RETURN(_SUCCESS)
+      end subroutine to_VerticalDimSpec
+
+      subroutine to_UngriddedDimsSpec(ungridded_dims_spec,attributes,rc)
+         type(UngriddedDimsSpec) :: ungridded_dims_spec
+         class(YAML_Node), intent(in) :: attributes
+         integer, optional, intent(out) :: rc
+
+         integer :: status
+         class(YAML_Node), pointer :: dim_specs, dim_spec
+         character(len=:), allocatable :: dim_name
+         integer :: dim_size,i
+         type(UngriddedDimSpec) :: temp_dim_spec
+
+         if (.not.attributes%has('ungridded_dim_specs')) then
+            _RETURN(_SUCCESS)
+         end if
+         dim_specs => attributes%of('ungridded_dim_specs')
+         do i=1,dim_specs%size()
+            dim_spec => dim_specs%of(i) 
+            call dim_spec%get(dim_name,'dim_name',_RC)
+            call dim_spec%get(dim_size,'extent',_RC)            
```

Comment:
One annoying unresolved issue is that I think some dimension extents may not be known at the time that this procedure runs.   I hope not.